src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -163,8 +161,38 @@ struct AggregateFunctionSumData
             Impl::add(sum, local_sum);
             return;
         }
+        else if constexpr (is_integer<T> || is_decimal<T>)
+        {
+            /// Use a mask to discard the value if it is null
+            T local_sum{};
+            using MaskType = std::conditional_t<sizeof(T) == 16, Int8, Int64>;
```

Comment:
These 2 lines need an explanation. Why & with a Int8 or Int64 vs the underlying type?